src/util/timemanager.cpp

```diff
@@ -412,6 +418,96 @@ void TimeManager::setDeltaTimeSteps(std::vector<double> deltaTimes) {
 
     _deltaTimeSteps = std::move(deltaTimes);
     _deltaTimeStepsChanged = true;
+
+    addDeltaTimesKeybindings();
+}
+
+void TimeManager::addDeltaTimesKeybindings() {
+    constexpr const std::array<Key, 10> Keys = {
+        Key::Num1,
+        Key::Num2,
+        Key::Num3,
+        Key::Num4,
+        Key::Num5,
+        Key::Num6,
+        Key::Num7,
+        Key::Num8,
+        Key::Num9,
+        Key::Num0
+    };
+
+    // First, clear any previous keybinds
+    for (const KeyWithModifier& kb : _deltaTimeStepKeybindings) {
+        global::keybindingManager.removeKeyBinding(kb);
+    }
+    _deltaTimeStepKeybindings.clear();
+    _deltaTimeStepKeybindings.reserve(_deltaTimeSteps.size());
+
+    // Find positive delta time steps
+    std::vector<double> steps;
+    const int nStepsGuess = static_cast<int>(std::floor(_deltaTimeSteps.size() * 0.5f));
+    steps.reserve(nStepsGuess);
+    std::copy_if(
+        _deltaTimeSteps.begin(),
+        _deltaTimeSteps.end(),
+        std::back_inserter(steps),
+        [](double value) { return value >= 0.0; }
+    );
+
+    const int nKeys = static_cast<int>(Keys.size());
+    const int nSteps = static_cast<int>(steps.size());
+    const int maxIterations = (nSteps >= nKeys) ? nKeys : nSteps;
+
+    const char* guiPath = DeltaTimeStepsKeybindsGuiPath;
```

Comment:
if you make this `constexpr const char*` you don't need to explicitly capture it in the lambda.    It might need to be `static`, too 🤔